src/main/java/de/muenchen/isi/domain/service/stammdaten/FoerdermixStammService.java

```diff
@@ -53,11 +54,17 @@ public FoerdermixStammModel getFoerdermixStammById(final UUID id) throws EntityN
      *
      * @param foerdermix zum Speichern.
      * @return das gespeicherte {@link FoerdermixStammModel}.
+     * @throws UniqueViolationException falls die Bezeichnung {@link FoerdermixStammModel#getBezeichnung()} des Fördermixes bereits im gleichen Jahr {@link FoerdermixStammModel#getBezeichnungJahr()} vorhanden ist.
      */
-    public FoerdermixStammModel saveFoerdermixStamm(final FoerdermixStammModel foerdermix) {
+    public FoerdermixStammModel saveFoerdermixStamm(final FoerdermixStammModel foerdermix) throws UniqueViolationException {
         var entity = this.stammdatenDomainMapper.model2Entity(foerdermix);
-        entity = this.foerdermixStammRepository.save(entity);
-        return this.stammdatenDomainMapper.entity2Model(entity);
+        final var saved = this.foerdermixStammRepository.findByBezeichnungJahrIgnoreCaseAndBezeichnungIgnoreCase(foerdermix.getBezeichnungJahr(), foerdermix.getBezeichnung());
```

Comment:
Durch eine Unique-Constraint (siehe https://github.com/it-at-m/isi-backend/pull/22/files#r1106878758) kann diese `findBy`-Logik entfallen, da in der Datenbank der Fehlerbereits abgesichert ist.

In der Methode ist dann die durch eine Verletzung des UniqueConstraint geworfenen Exception zu fangen und als UniqueViolationException weiterzugeben.

Reply:
angepasst